abusehelper.bots.archivebot: General code cleanup

jviide-work opened the pull request:
General code cleanup for `abusehelper.bots.archivebot`. Most notable things:
- Add documentation to `_encode_room_jid` docstrings.
- Underscore private functions.
- The commit message for a8f873707b34b1fd96dec16feaf929705124fafa is not complete, the commit also aims to simplify rotation logic.

jviide committed:
abusehelper.bots.archivebot: Add documentation

jviide committed:
abusehelper.bots.archivebot: Remove unnecessary flush() calls

jviide committed:
abusehelper.bots.archivebot: Close the opened fd if _unique_writable_file fails

jviide committed:
abusehelper.bots.archivebot: Underscore private functions

jviide committed:
abusehelper.bots.archivebot: Fix, call os.close before os.remove

mseppanen commented on abusehelper/bots/archivebot/archivebot.py:
```diff
@@ -85,11 +85,20 @@ def _unique_writable_file(directory, prefix, suffix):
 
-    with os.fdopen(fd, "wb") as fileobj:
+    try:
+        fileobj = os.fdopen(fd, "wb")
```
File descriptor `fd` is opened on line 79 without specifying `os.O_BINARY` (which is available only on Windows), however mode `"wb"`is specified for `fdopen()`. Not sure what is the end result of all this. This probably works just fine on UNIX systems, but it might do something weird on Windows.

jviide committed:
abusehelper.bots.archivebot: Set O_BINARY flag when using os.open

In _unique_writable_file the fd will be later wrapped into a file
object with fdopen(fd, "wb"). Therefor the fd should be opened with the
O_BINARY flag set (when applicable, i.e. on Windows).

mseppanen commented:
Looks good.
